Unix/protocol/protocol.c

```diff
@@ -2067,30 +2072,27 @@ static Protocol_CallbackResult _ProcessReceivedMessage(
                                 return PRT_RETURN_FALSE;
                             }
 
-                            // close socket to server
-                            trace_EngineClosingSocket(handler, handler->base.sock);
-                            Selector_RemoveHandler(socketAndBase->internalProtocolBase.selector, 
-                                                   &(socketAndBase->protocolSocket.base));
-
                             r = _ProtocolSocketTrackerRemoveElement(s);
                             if(MI_RESULT_OK != r)
                             {
                                 trace_TrackerHashMapError();
                                 return PRT_RETURN_FALSE;
                             }
+
+                            // close socket to server
+                            trace_EngineClosingSocket(handler, handler->base.sock);
                         }
                         else
                         {
                             forwardSock = handler->base.sock;
+                            ret = PRT_CONTINUE;
                         }
 
                         handler = newHandler;
 
-                        if(_SendAuthResponse(handler, binMsg->result, binMsg->authFile, forwardSock, 
-                                             binMsg->uid, binMsg->gid))
-                        {
-                            ret = PRT_CONTINUE;
-                        }
+                        if(!_SendAuthResponse(handler, binMsg->result, binMsg->authFile, forwardSock, 
+                                              binMsg->uid, binMsg->gid))
+                            trace_ClientAuthResponseFailed();
```

Comment:
Does OMI have a coding standard where all `if` statements should have surrounding braces?

Reply:
Added braces.